doc/qos/SONiC_ROCEv2_support.md

```diff
@@ -0,0 +1,574 @@
+     "dynamic_th": "3",
+     "pool": "[BUFFER_POOL|ingress_lossless_pool]",
+     "size": "0"
+ },
+ "pg_lossless_100000_40m_profile": {
+     "dynamic_th": "-3",
+     "pool": "[BUFFER_POOL|ingress_lossless_pool]",
+     "size": "1248",
+     "xoff": "177632",
+     "xon": "2288",
+     "xon_offset": "2288"
+ },
+ "pg_lossless_10000_40m_profile": {
+     "dynamic_th": "-3",
+     "pool": "[BUFFER_POOL|ingress_lossless_pool]",
+     "size": "1248",
+     "xoff": "37024",
+     "xon": "2288",
+     "xon_offset": "2288"
+ }
+},
+"BUFFER_PG": {
+       "Ethernet0|0": {
+           "profile": "[BUFFER_PROFILE|ingress_lossy_profile]"
+       },
+       "Ethernet0|3-4": {
+           "profile": "[BUFFER_PROFILE|pg_lossless_100000_40m_profile]"
+       },
+       ........
+       .......
+}
+
+"BUFFER_QUEUE": {
+  "Ethernet0|0-2": {
+          "profile": "[BUFFER_PROFILE|egress_lossy_profile]"
+   },
+  "Ethernet0|3-4": {
+          "profile": "[BUFFER_PROFILE|egress_lossless_profile]"
+   },
+  "Ethernet0|5-6": {
+          "profile": "[BUFFER_PROFILE|egress_lossy_profile]"
+  },
+  .......
+  .......
+}                                                                                                                                                                        48        563,7         19%
+
+```
+The above tables are populated by default already in the SONiC system upon executing
+the 'config qos buffer reload' Click command, we are providing the following configuration options to fine tune the values
+and the get support will be provided to dump the buffer information.
+
+### 1.1.3 Functional Requirements
+
+Provide management framework support to
+- Configure buffer pool and buffer pool
+- Associate PG & Queue with buffer profile
+
+### 1.1.4 Configuration and Management Requirements
+- CLI style configuration and show commands
+- REST API support
+- gNMI Support
+
+Details described in Section 3.
+
+### 1.1.6 Scalability Requirements
+
+### 1.1.7 Warm Boot Requirements
+
+## 1.2 Design Overview
+
+### 1.2.1 Basic Approach
+- Implement QoS buffer configuration and get support using transformer in sonic-mgmt-framework.
+
+### 1.2.2 Container
+The front end code change will be done in management-framework container including:
+- XML file for the CLI
+- Python script to handle CLI request (actioner)
+- Jinja template to render CLI output (renderer)
+- OpenConfig YANG model for QoS buffer openconfig-qos.yang
+- SONiC QOS buffer model on Redis DB schema
+- transformer functions to
+   * convert OpenConfig YANG model to SONiC YANG model for buffer related configurations
+
+### 1.2.3 SAI Overview
+
+# 2 Functionality
+
+## 2.1 Target Deployment Use Cases
+Manage/configure QoS buffer configurations via gNMI, REST and CLI interfaces
+
+## 2.2 Functional Description
+Provide CLI, gNMI and REST support for QoS buffer onfigurations.
+
+## 2.3 Backend change to support new configurations
+Provide change in management framework and buffermgrd modules.

```

Comment:
if possible please capture the changes required in buffermgrd .

Reply:
sure